### ISSUE TITLE
Implement kotlin compiler plugin for annotation processing

### ISSUE THREAD

**Issue by chippmann** [utopia-rise/kotlin-godot-wrapper#13]
**Describe the problem or limitation you are having in your project:**
As we need to process annotations without a jvm target. We need a compiler plugin to process the annotations and generate code from it.

**Describe how this feature / enhancement will help you overcome this problem or limitation:**
We can do the annotation processing without a fake jvm target (see #7 ).

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**
None to provide

**Describe implementation detail for your proposal (in code), if possible:**
Great sources around that topic:
https://youtu.be/w-GMlaziIyo
https://youtu.be/_obNBSldffw

**If this enhancement will not be used often, can it be worked around with a few lines of code?:**
Yes if the user has to write his registrations with json.

**Is there a reason why this should be in this project and not individually solved?:**
No as it is core functionality of this project.

**Comment by chippmann** [comment]
@piiertho I'm almost done with the compiler plugin. But one thing we need to discuss:
As i was developing it, i saw that many things we needed for this to work was already done in this project: https://github.com/Foso/MpApt (we talked about it on discord). There we said we'd rather not use it. But personally I changed my mind: Rather than reinventing the wheel, i would use the parts of it that we need to make annotation processing easier. Also if this project would die, i feel comfortable enough to say that i would be able to maintain it. Let alone that we already had to do this if we would implement that logic ourselves in this project.

To summarize: I would like to use the project we initially said we don't want to include, so we have to maintain less code. Should it die, I think i have enough knowledge to maintain the code we need it myself in this project.

One note: I can only finish once my [bufix](https://github.com/Foso/MpApt/pull/10) in said project is merged, as it currently crashes our sample builds. But given the activity of the maintainer, I guess this shouldn't take that long.

What do you think? Are you ok with that or should we maintain the needed code ourselves here?

**Comment by piiertho** [comment]
@chippmann, I also feel comfortable with maintaining it if the project dies too.
Let's use it as it does what we want.

**Comment by chippmann** [comment]
Compiler plugin is now done. I'm now waiting for the pull request https://github.com/Foso/MpApt/pull/10 to be merged and released.

**Comment by chippmann** [comment]
The pull request is merged and a new release was published: https://github.com/Foso/MpApt/releases/tag/0.8.4
I'll include it tomorrow and we should be good to go.
